### PR TITLE
Add support for `-enable-regex-literals` flag

### PR DESCRIPTION
**Pull request by artemcm**
Part of rdar://91119995

### CODE REVIEW

**Comment by artemcm** [comment]
Corresponding driver change in https://github.com/apple/swift-driver/pull/1045.

File: include/swift/Basic/LangOptions.h [hunk 1]
```diff
@@ -561,6 +561,9 @@ namespace swift {
     /// Enables dumping type witness systems from associated type inference.
     bool DumpTypeWitnessSystems = false;
 
+    /// Enables `/.../` syntax regular-expression literals
+    bool EnableForwardSlashRegexLiterals = false;
```
Comment (artemcm) [inline include/swift/Basic/LangOptions.h:565]: @hamishknight, anywhere where this can already be used to enable new parsing rules?

Comment (hamishknight) [reply include/swift/Basic/LangOptions.h:565]: Not currently no

**Review by hamishknight — APPROVED** [review]
Thanks!

**Comment by artemcm** [comment]
@swift-ci please test

**Review by nkcsgexi — COMMENTED** [review]
Nitpick: can we add a test this field is printed into `.swiftinterface`?

**Comment by artemcm** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@shahmishal - how come this didn't trigger the Windows CI?  This cased a regression which the CI would have caught.

**Comment by shahmishal** [comment]
> @shahmishal - how come this didn't trigger the Windows CI?  This cased a regression which the CI would have caught.

Not sure, we should make Windows required check

**Pull request merged**